tests/lava/proc/event_data/event_data_loader/test_aedat_data_loader.py

```diff
@@ -0,0 +1,295 @@
+        self.assertEqual(pm._shape_out, proc_params["shape_out"])
+        self.assertIsInstance(pm._file, AedatFile)
+        self.assertIsInstance(pm._stream,
+                              _AedatFileEventNumpyPacketIterator)
+        self.assertIsInstance(pm._frame_shape, tuple)
+        self.assertIsInstance(pm._random_rng, np.random.Generator)
+
+    def test_sub_sampling(self):
+        data_history = [
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+            [1, 1, 1, 1, 1, 1, 1, 1, 1],
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+            [1, 1, 1, 1, 1, 1],
+            [0],
+            [1, 1, 1],
+            [1],
+            [1],
+            [1]
+        ]
+        indices_history = [
+            [1597, 2308, 2486, 2496, 2498, 1787, 2642, 2633, 2489,
+             2488, 1596, 1729, 1727, 2500, 1780],
+            [1600, 1732, 2297, 1388, 2290, 2305, 3704, 3519, 1911],
+            [7138, 2301, 2471, 1601, 2982, 1364, 1379, 1386, 1384,
+             2983, 1390, 2289, 1401, 1362, 2293],
+            [1910, 1382, 1909, 1562, 1606, 1381],
+            [464],
+            [2323, 1908, 1393],
+            [4062],
+            [1792],
+            [3889]
+        ]
+        seed_rng = 0
+        rng = np.random.default_rng(seed=seed_rng)
+
+        max_num_events = 10
+        data_loader = AedatDataLoader(file_path="../dvs_recording.aedat4",
+                                      shape_out=(max_num_events,),
+                                      seed_sub_sampling=seed_rng)
+        recv_sparse = RecvSparse(shape=(max_num_events,))
+
+        data_loader.out_port.connect(recv_sparse.in_port)
+
+        # Run parameters
+        num_steps = 9
+        run_cfg = Loihi1SimCfg()
+        run_cnd = RunSteps(num_steps=1)
+
+        # Running
+        for i in range(num_steps):
+            data_loader.run(condition=run_cnd, run_cfg=run_cfg)
+
+            expected_data = np.array(data_history[i])
+            expected_indices = np.array(indices_history[i])
+
+            sent_and_received_data = \
+                recv_sparse.data.get()[:expected_data.shape[0]]
+            sent_and_received_indices = \
+                recv_sparse.idx.get()[:expected_indices.shape[0]]
+
+            if expected_data.shape[0] > max_num_events:
+                data_idx_array = np.arange(0, expected_data.shape[0])
+                sampled_idx = rng.choice(data_idx_array,
+                                         max_num_events,
+                                         replace=False)
+
+                expected_data = expected_data[sampled_idx]
+                expected_indices = expected_indices[sampled_idx]
+
+            np.testing.assert_equal(sent_and_received_data,
+                                    expected_data)
+            np.testing.assert_equal(sent_and_received_indices,
+                                    expected_indices)
+
+        # Stopping
+        data_loader.stop()
+
+    def test_run(self):
```

Comment:
You do not have to test basic running of a Process - that is tested more generally elsewhere. I would expect this test to run the Process and test that it works. This test does not check that the output is what you are expecting. Please add such an assertion.

Since this test then demonstrates the basic functionality, without subsampling, please pull it up to show up above the more specialized test that includes subsampling.